ipc/mdap.py

```diff
@@ -27,49 +28,81 @@ def connect_to_telemetry_server(address: tuple) -> socket.socket:
     print("Connection with the telemetry server established successfully.")
     return client_socket
 
+
+def test_telemetry(data_to_send, temperature):
+    # Convert the temperature from Kelvin to Celsius.
+    value = temperature - 273.15
+    app.logger.notice(f"BATTERY TEMP IS: {value}ºC")
+
+    timestamp = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+    if "WatchdogHeartbeatTvac" in data_to_send:
+        if "AdcTempKelvin" in data_to_send["WatchdogHeartbeatTvac"]["timestamp"]:
+            data_to_send["WatchdogHeartbeatTvac"][timestamp]["AdcTempKelvin"] = value
+    else:
+        data_to_send["data"]["WatchdogHeartbeatTvac"] = {
+            timestamp: {
+                "AdcTempKelvin": value
+            }
+        }
+
+
 def telem_to_message(data_to_send, payloads):
     for telemetry in payloads[TelemetryPayload]:
         telemetry = cast(TelemetryPayload, telemetry)
-        data_to_send[f"{telemetry.module.name}-{telemetry.channel.name}"] = {
-            "value": telemetry.data,
-            "timestamp": int(time.time()),
-            "bucket": "mission_data"
-        }
 
-        if (telemetry.module.name, telemetry.channel.name) == ('WatchdogHeartbeatTvac', 'AdcTempKelvin'):
-            app.logger.notice(f"BATTERY TEMP IS: {telemetry.data - 273.15}ºC")
-            data_to_send["TempC"] = {
-                "value": telemetry.data - 273.15,
-                "timestamp": int(time.time()),
-                "bucket": "mission_data"
-            }
+        populate_data_to_send(telemetry.module.name, telemetry.channel.name,
+                              telemetry.data, data_to_send)
+
+        # Testing against this particular module
+        if telemetry.module.name == "WatchdogHeartbeatTvac":
+            test_telemetry(data_to_send, telemetry.data)
```

Comment:
You pass telemetry.data here but the parameter is temperature

Reply:
`telemetry.data` is the temperature, right? 🤔  
I am just receiving it in a variable `temperature`

Reply:
Okay thats true, but I am confused about the check. Before we did:
`if (telemetry.module.name, telemetry.channel.name) == ('WatchdogHeartbeatTvac', 'AdcTempKelvin')`, but not you are only checking the module name. Its possible in this way that we could be adding values for other channels in this module as AdcTempKelvin.